addons/im_livechat/static/tests/chat_window_patch.test.js

```diff
@@ -72,6 +80,105 @@ test("closing a chat window with no message from admin side unpins it", async ()
+            livechat_operator_id: serverState.partnerId,
+            create_uid: serverState.publicUserId,
+        }))
+    );
+    setupChatHub({ folded: [channelIds[0], channelIds[1], channelIds[2], channelIds[3]] });
+    await start();
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "Visitor 5" });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 5" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 5…']:focus");
+    await withGuest(guestIds[1], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 2",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[1],
+            thread_model: "discuss.channel",
+        })
+    );
+    await withGuest(guestIds[1], () =>
+        rpc("/im_livechat/visitor_leave_session", { channel_id: channelIds[1] })
+    );
+    await withGuest(guestIds[3], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 4",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[3],
+            thread_model: "discuss.channel",
+        })
+    );
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 2 });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 4" });
+    await contains(".o-mail-Composer-input[placeholder='Message Visitor 4…']:focus");
+    await withGuest(guestIds[0], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 1",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[0],
+            thread_model: "discuss.channel",
+        })
+    );
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 3 });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 1" });
+    await withGuest(guestIds[2], () =>
+        rpc("/mail/message/post", {
+            post_data: {
+                body: "livechat 3",
+                message_type: "comment",
+                subtype_xmlid: "mail.mt_comment",
+            },
+            thread_id: channelIds[2],
+            thread_model: "discuss.channel",
+        })
+    );
+    await animationFrame();
+    await press("Tab");
+    await contains(".o-mail-ChatWindow", { count: 4 });
+    await contains(".o-mail-ChatWindow", { text: "Visitor 3" });
+    await animationFrame();
+    await press("Tab");
+    // Ensure the last tab selection is an ended livechat
+    await contains(".o-mail-ChatWindow", { text: "Visitor 2" });
+    await contains("span", { text: "This livechat conversation has ended" });
+});
```

Comment:
Same remark with the 2 other animation frames after `rpc`. The last `animationFrame()` is fine and can be kept.